src/test/groovy/com/cloudbees/plugins/flow/DSLTestCase.groovy

```diff
@@ -91,6 +78,30 @@ abstract class DSLTestCase extends HudsonTestCase {
         return flow.scheduleBuild2(0).get()
     }
 
+    def runWithAbortWhenWorseThanSuccess = { script ->
```

Comment:
please refactor, I think you only need one method `runWithAbortWhenWorseThan { script, abortWhenWorseThan -> ... }`